components/gitpod-db/src/typeorm/workspace-db-impl.ts

```diff
@@ -981,8 +987,28 @@ export abstract class AbstractTypeORMWorkspaceDBImpl implements WorkspaceDB {
      *       around to deleting them.
      */
     public async hardDeleteWorkspace(workspaceId: string): Promise<void> {
-        await (await this.getWorkspaceInstanceRepo()).update({ workspaceId }, { deleted: true });
-        await (await this.getWorkspaceRepo()).update(workspaceId, { deleted: true });
+        const prebuild = await this.findPrebuildByWorkspaceID(workspaceId);
+        if (prebuild !== undefined) {
+            // There are prebuilds linked to this workspace. We need to delete these first.
+            // `d_b_workspace_prebuild doens't have a deleted column for db-sync
+            // const prebuildsDeleted = await (
+            //     await this.getPrebuiltWorkspaceRepo()
+            // ).update({ id: prebuild.id }, { deleted: true });
+            // reportPrebuiltWorkspacePurged(prebuildsDeleted.affected || 0);
+
+            // Delete the d_b_prebuilt_workspace_updatable record?
+            // await (await this.getPrebuiltWorkspaceUpdatableRepo()).update()
```

Comment:
We don't have deletion by db-sync enabled for these. We'll need to enable this first.

Reply:
Current build reports: `src/typeorm/workspace-db-impl.ts(58,5): error TS6133: 'reportPrebuiltWorkspacePurged' is declared but its value is never read.`